src/test/java/com/wikia/webdriver/testcases/adstests/TestUrlBuilder.java

```diff
@@ -1,378 +1,420 @@
-    public void getUrlForPathTests() throws Exception {
-        Constructor<UrlBuilder> c = getUrlBuilderConstructor();
-        
-        Assert.assertEquals(c.newInstance("mediawiki119","preview", false, false, DEFAULT_LANGUAGE).getUrlForPath("/wiki/Special:Version"),
-               "http://mediawiki119.preview.wikia.com/wiki/Special:Version");
+  @Test(groups = {"TestUrlBuilder", "unitTests"})
+  public void getUrlForPathTests() throws Exception {
+    Constructor<UrlBuilder> c = getUrlBuilderConstructor();
 
-        String cb = "cb=1111";
-        Configuration.setTestValue("qs", cb);
+    Assert.assertEquals(c.newInstance("mediawiki119", "preview", false, false, DEFAULT_LANGUAGE)
+            .getUrlForPath("/wiki/Special:Version"),
+        "http://mediawiki119.preview.wikia.com/wiki/Special:Version");
 
-        Assert.assertEquals(c.newInstance("mediawiki119","prod", false, false, DEFAULT_LANGUAGE).getUrlForPath("Portal:Main"),
-                "http://mediawiki119.wikia.com/Portal:Main?" + cb);
-    }
+    String cb = "cb=1111";
+    Configuration.setTestValue("qs", cb);
 
-    @Test(groups = {"TestUrlBuilder", "unitTests"})
-    public void getUrlForPathWithLanguageTests() throws Exception {
-        Constructor<UrlBuilder> c = getUrlBuilderConstructor();
+    Assert.assertEquals(c.newInstance("mediawiki119", "prod", false, false, DEFAULT_LANGUAGE)
+            .getUrlForPath("Portal:Main"),
+        "http://mediawiki119.wikia.com/Portal:Main?" + cb);
+  }
 
-        Assert.assertEquals(c.newInstance("mediawiki119", "preview", false, true, "szl").getUrlForPath("/wiki/Special:Version"),
-                "http://mediawiki119.preview.wikia.com/szl/wiki/Special:Version");
+  @Test(groups = {"TestUrlBuilder", "unitTests"})
+  public void getUrlForPathWithLanguageTests() throws Exception {
+    Constructor<UrlBuilder> c = getUrlBuilderConstructor();
 
-        Assert.assertEquals(c.newInstance("mediawiki119", "preview", false, false, "szl").getUrlForPath("/wiki/Special:Version"),
-                "http://szl.mediawiki119.preview.wikia.com/wiki/Special:Version");
+    Assert.assertEquals(c.newInstance("mediawiki119", "preview", false, true, "szl")
+            .getUrlForPath("/wiki/Special:Version"),
+        "http://mediawiki119.preview.wikia.com/szl/wiki/Special:Version");
 
-    }
+    Assert.assertEquals(c.newInstance("mediawiki119", "preview", false, false, "szl")
+            .getUrlForPath("/wiki/Special:Version"),
+        "http://szl.mediawiki119.preview.wikia.com/wiki/Special:Version");
+  }
 
-    @Test(groups = {"TestUrlBuilder", "unitTests"})
-    public void appendQueryStringTests() throws Exception {
-        Constructor<UrlBuilder> c = getUrlBuilderConstructor();
-        UrlBuilder urlBuilder = c.newInstance("mediawiki119", "preview", false, false, DEFAULT_LANGUAGE);
+  @Test(groups = {"TestUrlBuilder", "unitTests"})
+  public void appendQueryStringTests() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00112)